Sites Dashboard: Add the primary action to open the site panel (Remove "click-on-row" behavior)

okmttdhr opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/9734
Also close https://github.com/Automattic/dotcom-forge/issues/9904

## Proposed Changes

This PR adds the primary action to open the site panel and remove the "click-on-row" behavior completely. 

- Removed the “row click” behavior from the table view
- Prepared the primary action to open the site panel
- Expanded the clickable area of the SiteField and moved the site URL and WP Admin links to actions. Also, added a hover color on the SiteField, but this will be revisited when we migrate Core's native primary/media field in https://github.com/Automattic/dotcom-forge/issues/9735.
- Added track events to know the trigger for the site panel opening. 

|view| before | after |
|--------|--------|--------|
|table view| <img width="1407" alt="Screenshot 2024-11-21 at 13 26 14" src="https://github.com/user-attachments/assets/dfe6fd27-00d9-4e68-a3a3-32ecbca8bdb2">the entire row is clickable | <img width="1307" alt="Screenshot 2024-11-21 at 13 25 42 copy 2" src="https://github.com/user-attachments/assets/f69038b1-15da-4b02-ac83-cecc575c6877"> |
|list view (when the panel is open)| <img width="407" alt="Screenshot 2024-11-21 at 13 26 59" src="https://github.com/user-attachments/assets/ace5e969-66e3-4673-92a1-9fb8e596992e">the entire row is clickable | <img width="403" alt="Screenshot 2024-11-21 at 13 27 09" src="https://github.com/user-attachments/assets/8cb9b83a-9aac-475c-88cf-cc2920299270">the entire row is clickable | 

Context: pfYzsZ-Mq-p2#comment-670

<details><summary>ver.1</summary>
<p>

<img width="1435" alt="Screenshot 2024-11-18 at 15 52 57" src="https://github.com/user-attachments/assets/a2bb2eca-4323-4ea1-a48d-32ff4ff74bf3">

</p>
</details> 


## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

Please refer to pbxlJb-6EH-p2#comment-4044 and https://github.com/Automattic/dotcom-forge/issues/9734 for more context. 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to /sites.
* Select any site to open the panel. You should click the ">" icon on the rightmost of the row. 
* Try to break this fly-out panel behavior.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

youknowriad committed:
Sites Dashboard: Replace actions field with dataviews actions

youknowriad committed:
Remove SiteEllipsisMenu component

youknowriad committed:
Replace some selectors usage

allilevine committed:
Change Settings action to Site settings.

youknowriad committed:
Show/hide actions properly depending active features

youknowriad committed:
Add new general settings action

youknowriad committed:
navigate using the router for calypso pages

youknowriad committed:
Remove action from fields

okmttdhr committed:
Merge branch 'trunk' into update/dashboard-actions

okmttdhr committed:
Merge branch 'trunk' into update/dashboard-actions

okmttdhr committed:
Merge branch 'trunk' into update/dashboard-actions

okmttdhr committed:
Add action menu open event

okmttdhr committed:
Override actions colors

okmttdhr committed:
Remove Share site for preview

okmttdhr committed:
Update list view style

okmttdhr committed:
Remove old comment

okmttdhr committed:
Add the primary action to open the site panel

okmttdhr committed:
Merge branch 'trunk' into update/dashboard-actions

okmttdhr committed:
Drop calypso_sites_dashboard_site_action_hosting_config_view

okmttdhr committed:
Moved color overrides into DataView wrapper

okmttdhr committed:
Fix client side rendering

okmttdhr committed:
Merge branch 'update/dashboard-actions' into update/dashboard-actions-primary

okmttdhr committed:
Ensures the row fits within the device width

okmttdhr committed:
Bring back isAdmin

okmttdhr committed:
Track site pane open

okmttdhr commented:
I implemented the chevron icon as the trigger to open the site panel, entirely removing the 'click-on-row' behavior.

One concern is that after opening the site panel, the cursor changes to a pointer on hover, which is Core's default behavior for the **list view**. This creates a slightly odd experience where the pointer suggests interactivity, but clicking the item does nothing. 

For now, I would suggest starting with the behavior in this PR and monitoring the metrics to see its impact. Based on the findings, we could consider iterating further by introducing an onRowClick prop pbxlJb-6EH-p2#improving-the-current-overrides in Core to handle this scenario better.

fushar reviewed (commented):
Hmmm..... I'm afraid users will find it harder to go to the hosting dashboard (site management panel) after this change. It seems we need to click the site icon precisely in order to trigger the flyout panel. I don't think users will notice the chevron icon 🤔 

BTW, I just noticed that Core's Pages list in the Site Editor has clickable row behavior. Has this been considered before? Do we know how Core achieves this?


<img width="706" alt="image" src="https://github.com/user-attachments/assets/7fa0b218-e227-458a-85f3-cf21e454e9f0">

matt-west commented:
Perhaps I missed something in reading through the context links, but it feels like this change significantly degrades the UX. It’s not obvious that clicking the site icon will open the fly-out panel, and the chevron icon can easily be missed.

As @fushar [pointed out above](https://github.com/Automattic/wp-calypso/pull/96388#pullrequestreview-2441952190), it seems like Core makes the entire row clickable unless multi-select is enabled on the table. A primary action is still displayed on hover, but the user can click anywhere on the row.

youknowriad commented:
Just noting that the screenshot above is the "list" layout, not the "table" layout. Table layout click behavior is always about selecting an item.

lucasmendes-design commented:
> A primary action is still displayed on hover, but the user can click anywhere on the row.

+1.

oandregal commented:
> BTW, I just noticed that Core's Pages list in the Site Editor has clickable row behavior. Has this been considered before? Do we know how Core achieves this?

The core component works the same for both list & table layouts: when the user clicks the row, it gets selected.

https://github.com/user-attachments/assets/436ee9b1-46ca-4775-8256-98c743612cf3

Note this is important for supporting bulk actions in the sites screen, which it doesn't now but I don't see why it shouldn't. For example, changing the language of several sites at once, or publish them, etc. There's more details at why we shouldn't override the "row click" behavior for table at pfYzsZ-g9.

okmttdhr committed:
Merge branch 'trunk' into update/dashboard-actions-primary

kangzj committed:
set usage stale time to 10 seconds (#96571)

madhusudhand committed:
Redirect back to media page post checkout (#96447)

* add redirect back to media page post checkout

* add upgrade status as query param

* fix video upload message

fushar committed:
Untangling: apply new design to Settings -> Administration -> Transfer site (#96527)

fushar committed:
Untangling: apply new design to Settings -> Administration -> Delete site (#96544)

okmttdhr committed:
Improved open-pane actions

cleacos committed:
Allow Pressable Referrals on staging (#96578)

bogiii committed:
Photos picker: Reduce polling interval from 5 to 3 seconds (#96547)

Commit:
Help Center: Improve dislike message for non eligible users (#96545)

gius80 committed:
Domain settings update for 100y domains (#96450)

* Update Domain Managment labels

* Update domain contact information card

* minor fix

* Add translator comments

escapemanuele committed:
Help Center: Switch to check if elapsed time less than 15 minutes (#96579)

bogiii committed:
Photos picker: Update the message in case when connection upgrade is required (#96580)

* Photos picker: Update the message in case when connection upgrade is required

j6ll committed:
Campaign details charts for Spend, Impressions and Clicks (#96346)

* Campaign details charts for Spend, Impressions and Clicks
* Make the graph respond to screensize
* Show spend stats - weekly if evergreen or total if deciduous
* Fix empty container borders showing when post has no stats yet (scheduled)
* Add overall spend when campaign is finished, and show total spend for active evergreen campaigns
* Rename method to avoid clash
* Refactor: Moves campaign stats query to own file
* typeguard the DSP apiVersion
* Move the chart to it's own folder, fix styling for the chart selector button (padding was missing on focus)
* Translate chart controls

heavyweight committed:
Fix error when clicking on link (#96582)

andregardi committed:
Create automated-migration/application-password feature flag. (#96562)

epeicher committed:
Business Tools: Refactor to extract component logic to a data component (#96542)

* Business Tools: Refactor to extract component logic to a data component

* Move translate and localizeUrl functions to main component

* Lift business logic to data component

* Move config logic to data component

Commit:
Help Center: Update copies (#96589)

sejas committed:
navigate to business checkout page to add the product by default (#96593)

madhusudhand committed:
rename sites/marketing/sharing route /sharing-buttons (#96581)

escapemanuele committed:
Site Migration flows should redirect to home instead of /start (#96591)

dognose24 committed:
Stats: Add Today and Yesterday to date range shortcuts (#96548)

* Add today and yesterday to shortcuts

* Gate new options behind the feature flag

* use hour as period for single days

---------

Co-authored-by: Jasper Kang <jasper.kang@automattic.com>

fushar committed:
Untangling: apply new design to Marketing -> {Traffic, Sharing} (#96586)

candy02058912 committed:
Update copy on Delete account screen (#96268)

* Update copy on Delete account screen

* Update Polldaddy to Crowdsignal

Co-authored-by: Aurorum <43215253+Aurorum@users.noreply.github.com>

---------

Co-authored-by: Aurorum <43215253+Aurorum@users.noreply.github.com>

candy02058912 committed:
Update copy for delete account confirmation modal (#96404)

candy02058912 committed:
Update delete link copy on account settings page (#96269)

fushar committed:
Untangling: apply new design to Settings -> Administration -> Manage connection (#96577)

okmttdhr committed:
Bring back the row click in the list view

okmttdhr committed:
"Open site" action

okmttdhr committed:
Add track events

okmttdhr commented on client/sites/components/sites-dataviews/index.tsx:
```diff
@@ -88,46 +91,25 @@ const DotcomSitesDataViews = ( {
 	//
 	// To do that, we need to pass a required `onSelectionChange` callback to signal that it is being used in controlled mode.
-	// However, when don't need to do anything in the callback, because we already maintain a selectedItem state.
 	// The current selection is a derived value which is [selectedItem.ID] (see getSelection()).
-	const onSelectionChange = () => {};
+	const onSelectionChange = useCallback(
+		( siteId: string ) => {
+			// In table view, when a row is clicked, the item is selected for a bulk action, so the panel should not open.
+			if ( dataViewsState.type !== 'list' ) {
+				return;
+			}
+			const site = sites.find( ( s ) => s.ID === Number( siteId ) );
+			if ( site ) {
+				openSitePreviewPane( site, 'list_row_click' );
+			}
+		},
+		[ dataViewsState.type, openSitePreviewPane, sites ]
+	);
```
Due to the use of `getItemId` on line 198, I implemented the "row-click" in the list view in this way and ended up adding `@ts-expect-error` on line 205. I'd appreciate your insights on whether there's a better way to do this. @lsl CC: @fushar

lsl replied:
I think we can drop that `item.id = item.ID.toString();` assignment from earlier now, per https://github.com/Automattic/wp-calypso/pull/96326/files#r1850274535

The [onSelectionChange function](https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/components/dataviews/index.tsx#L46) passes in a list of id's (this feature is for multiselection, not selecting a single item in the list like we're doing) so I think something like this should work:

```diff
@@ -93,12 +93,15 @@ const DotcomSitesDataViews = ( {
        // To do that, we need to pass a required `onSelectionChange` callback to signal that it is being used in controlled mode.
        // The current selection is a derived value which is [selectedItem.ID] (see getSelection()).
        const onSelectionChange = useCallback(
-               ( siteId: string ) => {
+               ( selectedSiteIds: string[] ) => {
                        // In table view, when a row is clicked, the item is selected for a bulk action, so the panel should not open.
                        if ( dataViewsState.type !== 'list' ) {
                                return;
                        }
-                       const site = sites.find( ( s ) => s.ID === Number( siteId ) );
+                       if ( selectedSiteIds.length === 0 ) {
+                               return;
+                       }
+                       const site = sites.find( ( s ) => s.ID === Number( selectedSiteIds[ 0 ] ) );
                        if ( site ) {
                                openSitePreviewPane( site, 'list_row_click' );
                        }
@@ -196,13 +199,10 @@ const DotcomSitesDataViews = ( {
                                selection={ getSelection() }
                                paginationInfo={ paginationInfo }
                                getItemId={ ( item ) => {
-                                       // @ts-expect-error -- From ItemsDataViews, this item.id assignation is to fix an issue with the DataViews component and item selection. It should be removed once the issue is fixed.
-                                       item.id = item.ID.toString();
                                        return item.ID.toString();
                                } }
                                isLoading={ isLoading }
                                defaultLayouts={ { table: {} } }
-                               // @ts-expect-error -- From ItemsDataViews, this item.id assignation is to fix an issue with the DataViews component and item selection. It should be removed once the issue is fixed.
                                onChangeSelection={ onSelectionChange }
                        />
                </div>
```

okmttdhr replied:
Thanks for the commit! Confirmed it's working.

okmttdhr committed:
Fix type error

lsl committed:
selectedSiteIds handling

fushar commented:
Wow this version is much better!

Would it make more sense for the site title and URL to be vertically center-aligned with the site icon? 
<img width="291" alt="image" src="https://github.com/user-attachments/assets/5c4ccb12-eefa-4d99-82b5-7a1ade39ecc4">

taipeicoder commented:
Couple of things I noticed:

1. The action icons being hidden when hovering over the column feels odds.

![Screen Capture on 2024-11-21 at 15-20-10](https://github.com/user-attachments/assets/f1aa1fe7-0004-4144-aeb4-3a1960286f05)

2. The pane icon still visible in the list view. Clicking it does nothing.

![Screenshot 2024-11-21 at 3 21 35 PM](https://github.com/user-attachments/assets/05b441e3-292c-4d42-979c-4ef158f9283c)

dognose24 committed:
Stats: Make chart bars navigable to a detailed chart (#96607)

* Replace the chart bar click with navigation to the detailed chart

* Stop actions when the period is hour

matt-west reviewed (commented):
Thanks @okmttdhr. @fushar and @taipeicoder already reported all the issues I found. once those are sorted, I think this is good to go.

Just a heads up that this interaction of clicking an action icon to trigger the fly-out will likely be removed once we transition to align fully with Core data views.

<img width="422" alt="Screenshot 2024-11-21 at 09 33 21" src="https://github.com/user-attachments/assets/19d64a0b-6919-4d7b-96ed-5842a7d4d660">

I’m collaborating with A4A, Jetpack, and Core on unifying our dashboard components and fully adopting all the behaviour of data views. I’ll post a P2 with more details soon.

escapemanuele committed:
Help Center: Do not call Support Interaction APIs in old version (#96540)

* Check for config

* Reenable useGetSupportInteractions

* Missing refactor

* Disable only on contact page

gmovr committed:
Update PULL_REQUEST_TEMPLATE.md (#96614)

Adding a new checklist item for whether changes have been tested in other languages.

rcrdortiz committed:
Fixed the upsells on premium themes (#96617)

* Fixed the upsells on premium themes

* Added the GS feature to the feature list panel of the plan upgrade

---------

Co-authored-by: Ricardo Ortiz <rcrd@MacBook-Pro-de-Ricardo.local>

epeicher committed:
Performance Profiler: Fix track event that sends llm survey results (#96611)

* Performance Profiler: Fix track event that sends llm survey results

* Remove description from track event as with chat_id we can obtain it

simison committed:
Substack Subscribers importer: release (#96528)

* Substack Subscribers importer: release

* disallow for jetpack cloud

---------

Co-authored-by: Miguel Lezama <lezama@gmail.com>

Commit:
Send Cart Key in Setup Intent Request (#96552)

* Send Cart Key in Setup Intent Request

* Added comment

j6ll committed:
Blaze location charts (#96511)

* Adds a breakdown of location data by country for impressions & clicks

* Get country names from Int, in user's locale

* Add ~0% for items below 0.01%, move campaign objectives into new box

Commit:
Preload setup page on the authorize page (#96558)

* Preload setup image

* Convert svg to react component

* Update client/jetpack-connect/woo-loader-stage-images/OpeningImage.tsx

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

* Update client/jetpack-connect/woo-loader.tsx

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

* Update client/jetpack-connect/woo-loader.tsx

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

* Fix lint issue

* Fix lint issue

---------

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

Commit:
Dynamically change color scheme from color_scheme query param (#96557)

* Change color theme by color_scheme query param

* Use --color-primary for the primary button and the progress bar

* Fix blue theme color to match the local WordPress

* Apply color_scheme from query

* Refresh color scheme when if it is woo passwordless JPC

* Return 'default' color scheme

* Remove blue color fix change -- no longer needed

* Fix error with isWooPasswordlessJPC access

CGastrell committed:
Stats: fix UTM upsell overlay link (#96596)

* use component or link for Learn more CTA, depending on env

* get rid of annoying notice due to lack of 'key' on iterated render

* pass supportContext for proper event tracking

Commit:
Update getPluginTitle to accept plugin names separated by , (#96468)

* Update function to accept plugin names separated by ,

* Fix broken tests

* Fix broken tests

* Fix bug where it always displays Jetpack regardless of plugin_name value

* Update client/lib/login/test/index.js

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

* Make sure plugin name is unique

* Rename variable

* Use Intl.ListFormat

* Fix broken test

* Remove langSlug requirement

* Fix broken test

* Return all available plugin names as a default value

---------

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

p-jackson committed:
Enable Bulk Plugin Management: Fix mobile layout and title alignment (#96576)

* Reinstate mobile layout for plugins page on JC and fix alignment

* Setting a limit for the header width so it remains aligned with the content

---------

Co-authored-by: Paulo Trentin <paulo@paulotrentin.com.br>

vykes-mac committed:
Logged-in Profiler: Public sites with missing launch status should be able to run performance test (#96559)

* when unlaunch site is empty check if the blog is public and allow performance profiling

* check launch status using blog_public setting

dhasilva committed:
Stats: Add list quick action to view generated URL on UTM modal (#96597)

* add utm data to utm card actions

* add default values for URL builder modal

* add view in URL builder quick action

sirbrillig committed:
Checkout: Add new PayPal payment method that uses PayPal JS (#94790)

* Experiment with adding new PayPal checkout payment method

* Remove createSubscription and other unsued things

* Add onClick to PayPalButtons

* Add payPalJsProcessor

* Add paypal-js processor to CheckoutProvider

* Replace PayPal_Express payment method name with PayPal_PPCP

* Add debug to paypal-js payment method

* Make sure that "payment method not available" error is displayed

* Do not render PayPalSubmitButton until it is active

* Adding the success and cancel URL to the transaction data

* Make new PPCP payment method creator function

* Restore WPCOM_Billing_PayPal_Express

* Restore NVP method in "add new card" form

* Create toggle for checkout to use NVP or PPCP

* Remove duplicate paypal-express key

* Add common fields to PayPal transaction response

* Add config option for paypal-ppcp

* Update yarn.lock

* Clean up payPalScriptOptions

* Add react-paypal-js to client/package.json

* Update yarn.lock again

* Revert checkout tests to use WPCOM_Billing_PayPal_Express

* Improve guard for paypal_order_id in processor

* Dedupe yarn.lock

* Call paypal-ppcp-confirm-payment on approval

* Add error handling for confirmation endpoint

* Allow creating PayPal Express even if ppcp is enabled

---------

Co-authored-by: Ellie Martin <ellie.martin@automattic.com>
Co-authored-by: Michael Cain <michael@michaeldcain.com>

matticbot committed:
fix(deps): update dependency @stripe/stripe-js to ^1.54.2 (#92660)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

paulopmt1 committed:
Make DataViews footer and plugin Management header sticky (#96592)

* Fixed DataViews footer and plugin Mangament header

* Implemented the same behavior for Jetpack Cloud

* Improved pagination so it wont conflict with dev badge

* Limit how much DataViews content will expand

* Removed unnecessary style

travisw committed:
make sure we are not trying to use an undefined plan (#96652)

okmttdhr committed:
Hide Overview action when the site is selected

okmttdhr committed:
Don't cut off the focus outline

okmttdhr committed:
Align SiteField vertically

okmttdhr commented:
Thanks for your review, @fushar @taipeicoder @matt-west @lucasmendes-design!

>Would it make more sense for the site title and URL to be vertically center-aligned with the site icon?
>(Minor) Focus state being cut off on the right side.

This has been addressed!

>The action icons being hidden when hovering over the column feels odd.

I’ve fixed this in a separate PR: https://github.com/Automattic/wp-calypso/pull/96608.

>The pane icon is still visible in the list view. Clicking it does nothing.

Good catch! I’ve addressed this by hiding the icon when the site is selected. Note that the "Open site" action is now displayed for the selected site, which I believe makes sense.

| no state | hover | selected |
|--------|--------|--------|
|<img width="401" alt="Screenshot 2024-11-22 at 10 56 24" src="https://github.com/user-attachments/assets/c5b916f7-d339-44b7-b4a8-7a426647203e">| <img width="400" alt="Screenshot 2024-11-22 at 10 55 35" src="https://github.com/user-attachments/assets/a622da79-7250-49ef-b91c-e9dc201cfdb0"> | <img width="399" alt="Screenshot 2024-11-22 at 10 55 28" src="https://github.com/user-attachments/assets/2de92858-cf10-4db2-b747-d50569a575b1"> | 

>(Minor) Layout jump when switching focus between Stats and Actions.

This turned out to be less straightforward to resolve. The issue arises due to:

- Dropping the fixed table layout in https://github.com/Automattic/wp-calypso/pull/96221.
- The Stats column takes up the full width of the `<td>`.

To address this, I suggest we follow up by replacing the current Stats column with A4A’s implementation, which seems more appropriate. pbxlJb-6A9-p2#comment-4018. I’ll create an issue for this.

okmttdhr committed:
Merge branch 'trunk' into update/dashboard-actions-primary

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17012155).

oandregal commented on client/sites/components/sites-dataviews/actions.tsx:
```diff
@@ -13,19 +15,90 @@ import {
+					}
+				},
+				isEligible: ( site ) => {
+					if ( site.ID === selectedItem?.ID ) {
```
Judging by the code, I presume this is for hiding the `site-overview` action in the list view. This works, but it also runs the logic for every item. An alternative would be to pass the `view.type` to the useActions hook and only add `site-overview` to the actions array if the view is table. It would also not work well when/if the sites view adds support for bulk actions.

oandregal replied:
@okmttdhr actually, this is what happens (note how the actions changes after clicking):

https://github.com/user-attachments/assets/8afc1fb2-ef05-4a4a-9e2b-a81113e8073b

okmttdhr replied:
Thank you for your review!

I agree that hiding the Overview primary action in the list view makes sense—it ensures consistency, and the row-click interaction is provided for the list view. I created https://github.com/Automattic/wp-calypso/pull/96715 to address this.
